x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -184,6 +209,20 @@ func (c *enrollCmd) Execute(ctx context.Context) error {
 			errors.M(errors.MetaKeyURI, c.options.URL))
 	}
 
+	if localFleetServer {
+		// Ensure that the agent does not use a proxy configuration
+		// when connecting to the local fleet server.
+		c.remoteConfig.Transport.Proxy.Disable = true
```

Comment:
How does this persist after enrollment? This looks like it will work for enrollment, but how does the restarted Elastic Agent after enrollment ensure that it talks to its local Fleet Server not over the proxy? Does this `Disable = true` get persisted into the `fleet.yml` so that Elastic Agent knows not to use the proxy?

Reply:
Updated PR. The config for agent is serialized in `fleetServerBootstrap`.